Vitaly sprint1

lump1k opened the pull request:
# Pull request for issue: #<insert relevant github issue numbers that this PR is related to>

This is a pull request for the following functionalities:

<Insert description of the proposed new functionalities and how they can be tested>

## Changes to the database schema

<Insert description of changes to database schema>

## Changes to the model classes

<Insert description of changes to model classes>

## Changes to the controller classes

<Insert description of changes to controller classes>

## Changes to the view template files

<Insert description of changes to view template files>

## Changes to the style

<Insert description of changes to the look and feel of any web pages>

## Changes to the tests

<Insert description of changes to the test code>

## Changes to the provisioning

<Insert description of any changes related to the building of the GigaDB application>

pli888 commented:
Hello @lump1k To help me approve this pull request, is it possible for you to describe the work you have done please? You can take a look at this recent [pull request](https://github.com/gigascience/gigadb-website/pull/273) to see how a PR can be described.

In addition, can you provide commit messages when you check in code?

Where possible, it is useful to reference commits and pull requests to Github issues so I can see what work relates to which user requirements.

Without this information, it is difficult for me and others to understand what you have done and the reasons for your code changes.

lump1k commented:
i just added everything so u can test if u can run tests. because i have errors.
maybe i should create pull request to another branch in this case?
i understand that it will be better if i create 1 pull request for 1 task. but as Peter wanted to check why i cant run tests

lump1k commented:
/datasetSubmission/index
Dataset submission
/datasetSubmission/upload
Upload your dataset metadata from a spreadsheet
/datasetSubmission/study
In Authors Tab should work posibility to add Authors by csv or tsv files
For CrediT i create a table in database contribution
In Additional information Tab
I added column regexp to table prefix. So it should validate value by pattern from this column
for others links I added these patterns:

case self::MANUSCRIPTS:
return '/^doi:[0-9]+\.[0-9]+\/gigascience\/[a-z]+[0-9]+$/i';
case self::PROTOCOLS:
return '/^doi:[0-9]+\.[0-9]+\/protocols\.io\.[a-z0-9]+$/i';
case self::_3D_IMAGES:
return '/^https:\/\/skfb\.ly\/[a-z0-9]+$/i';
case self::CODES:
return '/^<script[\s\S]*?>[\s\S]*?<\/script>$/i';
default:
return '/^doi:[0-9]+\.[0-9]+\/[0-9]+\.[0-9]+$/i';
